## items to check

- number of articles
- number of authors
- check under-splitting: different authors share the same id -- id labeling mistake
- check over-splitting: the same author have different author_ids

In [34]:
import pandas as pd

In [19]:
my_data = pd.read_csv("C:\\Users\\yuanx\\Documents\\GitHub\\author_network_analysis\\Salt\\data\\salt_cont_author.csv", 
                      na_values="NA", encoding="UTF-8")

In [35]:
my_data.head(5)

ID                                              title author_given_name  \
0   1  Systematic review of long term effects of advi...               Lee   
1   1  Systematic review of long term effects of advi...       Christopher   
2   1  Systematic review of long term effects of advi...      George Davey   
3   1  Systematic review of long term effects of advi...              Shah   
4   2  Reduced dietary salt for prevention of cardiov...                L.   

  author_surname    author_id organizational_author  manual  
0         Hooper  57202855325                   NaN   False  
1       Bartlett  57194377301                   NaN   False  
2          Smith  35354080500                   NaN   False  
3        Ebrahim   7102093023                   NaN   False  
4         Hooper  57202855325                   NaN   False

In [36]:
len(my_data.ID.sort_values().unique()) # total article 82

82

In [37]:
len(my_data.author_id.sort_values().unique()) # total authors 593

593

## Check under-splitting

In [38]:
full_name = my_data.author_given_name + " " + my_data.author_surname
name_id_df = pd.DataFrame(data = {'full_name': my_data.author_given_name + " " + my_data.author_surname,
                                 'author_id': my_data.author_id})

In [43]:
name_id_df

full_name    author_id
0              Lee Hooper  57202855325
1    Christopher Bartlett  57194377301
2      George Davey Smith  35354080500
3            Shah Ebrahim   7102093023
4               L. Hooper  57202855325
..                    ...          ...
778            Ivan Mills          108
779           Margo Woods          110
780        Elaine Blethen          112
781           Ann LaBrode          199
782       Andrea Dommeyer          107

[783 rows x 2 columns]

In [41]:
len(name_id_df.drop_duplicates())

627

In [56]:
len(name_id_df.drop_duplicates())

627

In [ ]:
check_df_1 = name_id_df.groupby("author_id").count()
check_df_1[check_df_1['full_name']>1].rename(columns={"full_name": "no_of_names"}).reset_index().merge(name_id_df, on='author_id').drop_duplicates().to_csv("under_splitting_checking.csv", index=False)

## Correct under-splitting

Discovered two under-splitting: 32 and 33

32	3	Jeannie McKenzie  
32	3	N. Carole Milas

33	3	Robert W. Jeffery  
33	3	Monica E. Yamamoto  

33	3	Monica Yamamoto  

### corrected them mannually  
- Jeannie McKenzie change from 32 to 200  
- Robert W. Jeffery change from 33 to 201

## Check over-splitting

In [57]:
check_df_2 = name_id_df.drop_duplicates().groupby("full_name").count().sort_values(by='author_id', ascending=False)

In [58]:
check_df_2.rename(columns={"author_id": "no_of_ids"},inplace=True)

In [59]:
check_df_2[check_df_2['no_of_ids']>1].reset_index().merge(name_id_df, on='full_name').drop_duplicates()

full_name  no_of_ids    author_id
0  George Davey Smith          2  35354080500
1  George Davey Smith          2  57220145221
2        Shah Ebrahim          2   7102093023
3        Shah Ebrahim          2  56500020000
4     Paul K. Whelton          2   7103041111
5     Paul K. Whelton          2   7102422991

## Over-splitting correction

For each over splitting, replace the larger author_id with smallest author_id

George Davey Smith: all 57220145221 to 35354080500  
Sah Ebrahim: all 56500020000 to 7102093023  
Paul K. Whelton: all 7102422991 to 7103041111



## Final check after corrections

In [60]:
my_data = pd.read_csv("C:\\Users\\yuanx\\Documents\\GitHub\\author_network_analysis\\Salt\\data\\salt_cont_author.csv", 
                      na_values="NA", encoding="UTF-8")

full_name = my_data.author_given_name + " " + my_data.author_surname
name_id_df = pd.DataFrame(data = {'full_name': my_data.author_given_name + " " + my_data.author_surname,
                                 'author_id': my_data.author_id})

In [63]:
check_df_1 = name_id_df.groupby("author_id").count()
check_df_1[check_df_1['full_name']>1].rename(columns={"full_name": "no_of_names"}).reset_index().merge(name_id_df, on='author_id').drop_duplicates().to_csv("after_correction.csv", index=False)

In [62]:
check_df_2 = name_id_df.drop_duplicates().groupby("full_name").count().sort_values(by='author_id', ascending=False)
check_df_2.rename(columns={"author_id": "no_of_ids"},inplace=True)
check_df_2[check_df_2['no_of_ids']>1].reset_index().merge(name_id_df, on='full_name').drop_duplicates()

Empty DataFrame
Columns: [no_of_ids, full_name, author_id]
Index: []